### PR TITLE
Add teletubbies feature

### PR DESCRIPTION
**Pull request by whitecollin**
- add title
- add teletubbie list
- add search bar
- add scroll event

### CODE REVIEW

File: src/pages/Teletubbies/index.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import { getTeletubbies } from 'api'
+
+const Teletubbies = () => {
+  const [teletubbies, setTeletubbies] = useState([])
+  const [offset, setOffset] = useState(0)
+  const [searchWord, setSearchWord] = useState('')
+  const handleSearch = event => {
+    setSearchWord(event.target.value.toLowerCase())
+  }
+  useEffect(() => {
```
Comment (codeLover1126) [inline src/pages/Teletubbies/index.js:14]: To make it more readable, please insert a new blank line.

File: src/pages/Teletubbies/index.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      (async function () {
+        const teletubbies = await getTeletubbies()
+        setTeletubbies(teletubbies)
+      })()
+    }
+    catch(err){
+      throw err;
+    }
+  }, []);
+  useEffect(() => {
```
Comment (codeLover1126) [inline src/pages/Teletubbies/index.js:25]: A blank line here.

File: src/pages/Teletubbies/index.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      ) {
+        setOffset(offset+20);
+      }
+    }
+    window.addEventListener('scroll', handleScroll)
+    return () => {
+      window.removeEventListener('scroll', handleScroll)
+    }
+  }, [offset])
+  return (
```
Comment (codeLover1126) [inline src/pages/Teletubbies/index.js:39]: A blank line as well.

File: src/pages/Teletubbies/index.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    catch(err){
+      throw err;
+    }
+  }, []);
+  useEffect(() => {
+    const handleScroll = () => {
+      if (
+        window.innerHeight + window.scrollY >=
+        document.body.offsetHeight
```
Comment (codeLover1126) [inline src/pages/Teletubbies/index.js:29]: It's okay to unwrap the line here.

File: src/pages/Teletubbies/index.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  }
+  useEffect(() => {
+    try{
+      (async function () {
+        const teletubbies = await getTeletubbies()
+        setTeletubbies(teletubbies)
+      })()
+    }
+    catch(err){
+      throw err;
```
Comment (codeLover1126) [inline src/pages/Teletubbies/index.js:22]: Can you add the logic for handling an error? Like displaying the error message on the page. What do you think?

**Review by codeLover1126 — COMMENTED** [review]
Nice work! Please review comments though.

File: src/pages/Teletubbies/index.js (hunk 1 above)
Comment (whitecollin) [reply src/pages/Teletubbies/index.js:20]: fixed it

File: src/pages/Teletubbies/index.js (hunk 2 above)
Comment (whitecollin) [reply src/pages/Teletubbies/index.js:32]: fixed it

File: src/pages/Teletubbies/index.js (hunk 3 above)
Comment (whitecollin) [reply src/pages/Teletubbies/index.js:64]: fixed it

File: src/pages/Teletubbies/index.js (hunk 4 above)
Comment (whitecollin) [reply src/pages/Teletubbies/index.js]: fixed it

File: src/pages/Teletubbies/index.js (hunk 5 above)
Comment (whitecollin) [reply src/pages/Teletubbies/index.js]: fixed it

**Review by codeLover1126 — CHANGES_REQUESTED** [review]
NEVER repeat same errors.

File: src/pages/Teletubbies/index.js [hunk 6]
```diff
@@ -47,6 +72,13 @@ const Teletubbies = () => {
         onChange={handleSearch}
         label='Search by name'
         variant='outlined' />
+        <Snackbar
+        open={alert}
```
[the remark this answers is not in the document]
Comment (whitecollin) [reply src/pages/Teletubbies/index.js:76]: fixed it

[the remark this answers is not in the document]
Comment (whitecollin) [reply src/pages/Teletubbies/index.js:76]: fixed it

**Comment by codeLover1126** [comment]
Well done.

**Pull request merged**